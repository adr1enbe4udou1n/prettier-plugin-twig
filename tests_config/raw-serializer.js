const RAW = Symbol.for("raw");

const print = val => val[RAW];
const test = val =>
    val &&
    Object.prototype.hasOwnProperty.call(val, RAW) &&
    typeof val[RAW] === "string";
export default {
    print,
    test
};
