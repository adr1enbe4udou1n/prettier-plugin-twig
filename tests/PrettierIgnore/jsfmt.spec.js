import { run_spec } from "tests_config/run_spec";
import { describe, expect, it } from "vitest";

describe("Prettier ignore", () => {
    it("should handle single line ignore", async () => {
        const { actual, snapshotFile } = await run_spec(import.meta.url, {
            source: "prettierIgnore.twig"
        });
        await expect(actual).toMatchFileSnapshot(snapshotFile);
    });
    it("should handle ignore blocks", async () => {
        const { actual, snapshotFile } = await run_spec(import.meta.url, {
            source: "prettierIgnoreStartEnd.twig"
        });
        await expect(actual).toMatchFileSnapshot(snapshotFile);
    });
});
