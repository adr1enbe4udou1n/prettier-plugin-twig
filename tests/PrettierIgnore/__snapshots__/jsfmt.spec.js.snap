// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`prettierIgnore.twig - twig-verify > prettierIgnore.twig 1`] = `
<!-- prettier-ignore -->
<div><span    class="what-about-this"  >Should not re-format</span></div>
<div><span    class="what-about-this"  >Should re-format</span></div>

{#prettier-ignore#}
<div><span    class="what-about-this"  >Should not re-format</span></div>
<div><span    class="what-about-this"  >Should re-format</span></div>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<!-- prettier-ignore -->
<div><span    class="what-about-this"  >Should not re-format</span></div>
<div>
    <span class="what-about-this">Should re-format</span>
</div>

{# prettier-ignore #}
<div><span    class="what-about-this"  >Should not re-format</span></div>
<div>
    <span class="what-about-this">Should re-format</span>
</div>

`;

exports[`prettierIgnoreStartEnd.twig - twig-verify > prettierIgnoreStartEnd.twig 1`] = `
<!--prettier-ignore-start-->
<div><span    class="what-about-this"  >Should not re-format</span></div>
<div><span    class="what-about-this"  >Should not re-format</span></div>
<div><span    class="what-about-this"  >Should not re-format</span></div>
<!-- prettier-ignore-end -->

<div><span    class="what-about-this"  >Should re-format</span></div>

{# prettier-ignore-start #}
<div><span    class="what-about-this"  >Should not re-format</span></div>
<div><span    class="what-about-this"  >Should not re-format</span></div>
<div><span    class="what-about-this"  >Should not re-format</span></div>
{# prettier-ignore-end #}

<div><span    class="what-about-this"  >Should re-format</span></div>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<!-- prettier-ignore-start -->
<div><span    class="what-about-this"  >Should not re-format</span></div>
<div><span    class="what-about-this"  >Should not re-format</span></div>
<div><span    class="what-about-this"  >Should not re-format</span></div>
<!-- prettier-ignore-end -->

<div>
    <span class="what-about-this">Should re-format</span>
</div>

{# prettier-ignore-start #}
<div><span    class="what-about-this"  >Should not re-format</span></div>
<div><span    class="what-about-this"  >Should not re-format</span></div>
<div><span    class="what-about-this"  >Should not re-format</span></div>
{# prettier-ignore-end #}

<div>
    <span class="what-about-this">Should re-format</span>
</div>

`;
